src/domain/opendatafabric/src/identity/dataset_refs.rs

```diff
@@ -821,4 +842,22 @@ impl std::str::FromStr for DatasetRefPattern {
     }
 }
 
-super::dataset_identity::impl_parse_error!(DatasetRefPattern);
+impl std::str::FromStr for DatasetRefPatternAny {
+    type Err = ParseError<Self>;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if s.contains('%') {
+            return match DatasetPattern::from_str(s) {
+                Ok(dataset_pattern) => Ok(Self::Pattern(dataset_pattern)),
+                Err(_) => Err(Self::Err::new(s)),
+            };
+        }
+        match DatasetRefAny::from_str(s) {
+            Ok(dataset_ref) => Ok(Self::RefAny(dataset_ref)),
+            Err(_) => Err(Self::Err::new(s)),
+        }
+    }
+}
```

Comment:
This style of parsing and `DatasetPattern` model is missing a case of multi-tenant remote alias:
   `RepoName/AccountName/DatasetNamePattern`

I guess we might need `DatasetRemotePattern` to cut off both names before dataset name pattern.